miden/src/cli/data.rs

```diff
@@ -80,33 +103,56 @@ impl OutputFile {
         Ok(outputs)
     }
 
-    /// write the output file
+    /// Write the output file
     pub fn write(outputs: Vec<u64>, path: &Option<PathBuf>) -> Result<(), String> {
-        match path {
-            Some(path) => {
-                // if path provided create output file
-
-                println!("Creating output file `{}`", path.display());
-
-                let file = fs::File::create(&path).map_err(|err| {
-                    format!(
-                        "Failed to create output file `{}` - {}",
-                        path.display(),
-                        err
-                    )
-                })?;
-
-                println!("Writing data to output file");
-
-                // write outputs to output file
-                serde_json::to_writer_pretty(file, &Self { outputs })
-            }
-
-            // no path provided - write outputs to stdout
-            None => serde_json::to_writer_pretty(std::io::stdout(), &Self { outputs }),
+        if let Some(path) = path {
+            // if path provided, create output file
+            println!("Creating output file `{}`", path.display());
+
+            let file = fs::File::create(&path).map_err(|err| {
+                format!(
+                    "Failed to create output file `{}` - {}",
+                    path.display(),
+                    err
+                )
+            })?;
+
+            println!("Writing data to output file");
+
+            // write outputs to output file
+            serde_json::to_writer_pretty(
+                file,
+                &Self {
+                    outputs: outputs
+                        .iter()
+                        .map(|v| v.to_string())
+                        .collect::<Vec<String>>(),
+                },
+            )
+        } else {
+            println!(
+                "Output: {}",
+                format!(
+                    "{}",
+                    outputs
+                        .iter()
+                        .map(|v| v.to_string())
+                        .collect::<Vec<String>>()
+                        .join(" ")
+                )
+            );
+            Ok(())
         }
         .map_err(|err| format!("Failed to write output data - {}", err))
     }
+
+    /// Converts outputs vector of String to vector of u64
+    pub fn outputs_u64(&self) -> Vec<u64> {
```

Comment:
nit: I think it is OK to name this just as `outputs()` (without the `u64` suffix).